wolfcrypt/src/asn.c

```diff
@@ -6122,6 +6122,14 @@ static WC_INLINE int GetTime(int* value, const byte* date, int* idx)
 int ExtractDate(const unsigned char* date, unsigned char format,
                                                   struct tm* certTime, int* idx)
 {
+    /* Extract the time from the struct tm - 16bit processors store as uint8_t */
```

Comment:
@akmcomau , these changes will not return the parsed time in `certTime`. You'll need to make sure and populate `certTime` at the end of the function. Also the original code called `GetTime` first then offset  year by 1900. Is there a reason you can't keep that behavior? Same for month.

Reply:
This adjustment is made on lines 6147 & 6148 ... but I can move it up here.

Reply:
If you look carefully the `certTime` computed value is not returned. Each element is parsed, but its never put back into `certTime`.